addons/spreadsheet_dashboard/models/spreadsheet_dashboard.py

```diff
@@ -10,3 +13,14 @@ class SpreadsheetDashboard(models.Model):
     dashboard_group_id = fields.Many2one('spreadsheet.dashboard.group', required=True)
     sequence = fields.Integer()
     group_ids = fields.Many2many('res.groups', default=lambda self: self.env.ref('base.group_user'))
+
+
+    def fetch_data(self):
```

Comment:
could we maybe rename "fetch_data" in something more explicit that indicates that we are in readonly ? (to help differentiate it from its edition equivalent)